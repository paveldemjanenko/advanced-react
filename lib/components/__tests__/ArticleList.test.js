import React from 'react';
import ArticleList from '../ArticleList';
// import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
// import Article from '../Article';

Enzyme.configure({ adapter: new Adapter() });

describe('ArticleList', () => {

  const testProps = {
    articles: {
      a: { id: 'a' },
      b: { id: 'b' },
    },
    // store: {
    //   lookupAuthor: jest.fn(() => ({})),
    // }
  };

  // Article.propTypes = {};
  
  it('render correctly', () => {
    // const tree = renderer.create(
    //   <ArticleList 
    //     {...testProps}
    //   />
    // ).toJSON();

    const wrapper = shallow(
      <ArticleList 
        {...testProps}
      />
    );

    expect(wrapper.find('ArticleContainer').length).toBe(2);
    expect(wrapper).toMatchSnapshot();

    // expect(tree.children.length).toBe(2);
    // expect(tree).toMatchSnapshot();
  });

});
