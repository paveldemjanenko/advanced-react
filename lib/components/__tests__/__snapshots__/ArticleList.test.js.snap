// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList render correctly 1`] = `ShallowWrapper {}`;
